### PR TITLE
SB-46: Get reservations data from Firebase

### PR DESCRIPTION
**Pull request by lubualo**
- Create a service to handle the communication with Firebase for reservations-related operations.
- Define the data structure for reservations in Firebase
- Come up with the reservation class/model/representation as object in the frontend.
- A temporary page was created to visualize the results from the query
image

### CODE REVIEW

**Comment by lubualo** [comment]
[Get/Set Court Reservations in Firebase](https://trello.com/c/T0QhLWBt/46-get-set-court-reservations-in-firebase)

File: app/firebase/reservations/reservation.ts [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+	);
+	const docSnap = await getDoc(docRef);
+	if (!docSnap.exists()) {
+		throw new Error('The reservation does not exist.');
+	}
+	const data = { ...docSnap.data() };
+	return {
+		court: data.court,
+		owner: data.owner,
+		duration: data.duration,
```
Comment (EzeLamar) [inline app/firebase/reservations/reservation.ts:22]: The firebase service should be in charge of calculate the `finalTime` based on `initialTime` and `duration` fields.

You can use `moment` library, which was already added to the project.

Here is an example of how you can implement it:
```suggestion
	finalTime: moment(data.initialTime.toDate()).add(data.duration, 'minutes').toDate(),	
```

File: app/firebase/reservations/reservation.ts [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+		initialTime: data.initialTime.toDate(),
+	};
+}
+
+// export async function editReservation(
```
Comment (EzeLamar) [inline app/firebase/reservations/reservation.ts:27]: please remove this commented function

**Review by EzeLamar — APPROVED** [review]
Amazing job! 🚀

File: app/firebase/reservations/reservation.ts (hunk 1 above)
Comment (EzeLamar) [reply app/firebase/reservations/reservation.ts:22]: remember to rebase to origin/master to get access to the moment library 😄

Comment (lubualo) [reply app/firebase/reservations/reservation.ts:22]: I'm wondering if this should be part of the model or just the representation in the UI Component?

Comment (EzeLamar) [reply app/firebase/reservations/reservation.ts]: I agree with you. For now we can leave it as it is to move forward with the development. We will review the current design later.

**Pull request merged**